pkg/logs/sender/formatter.go

```diff
@@ -0,0 +1,62 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package sender
+
+import (
+	"bytes"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+)
+
+// Formatter transforms a batch of messages into a payload.
+type Formatter interface {
+	Format(messages []*message.Message) []byte
+}
+
+var (
+	// LineFormatter is a shared line formatter.
```

Comment:
More general question: Here we are creating instances of formatters shared by the senders. It's fine as they are stateless but if someone starts to add a state to the formatter it's going to be dangerous to share those instances.

What if we were a bit more defensive by instantiating those when we create the pipeline (when we pick the strategy?). The same comment applies for the StreamStrategy.

Reply:
If someone does such a change, I expect him to wonder what are the impact of such a change. I'll stick with shared instances.

Reply:
Updated Serializers, kept StreamStrategy